### ISSUE TITLE
ARM doesn't support DMI calls

### ISSUE THREAD

**Issue by philmmanjaro** [manjaro/mdd#1]
We should decide on how we add other architectures like `aarch64` from ARM64. There DMI calls don't work as there is no BIOS given. An example error message from Ray trying to run `mmd` on one of his devices:

```
mdd --dry-run
Welcome to MDD - The Manjaro Data Donor
Preparing data submission...
cat: /sys/devices/virtual/dmi/id/product_name: No such file or directory
cat: /sys/devices/virtual/dmi/id/product_family: No such file or directory
cat: /sys/devices/virtual/dmi/id/sys_vendor: No such file or directory
cat: /sys/devices/virtual/dmi/id/board_name: No such file or directory
Traceback (most recent call last):
File "/usr/bin/mdd", line 744, in <module>
main()
File "/usr/bin/mdd", line 709, in main
data = get_device_data(args.telemetry)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/bin/mdd", line 651, in get_device_data
"cpu": get_cpu_info(),
^^^^^^^^^^^^^^
File "/usr/bin/mdd", line 259, in get_cpu_info
[
IndexError: list index out of range
```

Some `inxi` sample outputs:

```
inxi -Cxx
CPU:
  Info: quad core model: N/A variant: cortex-a76 bits: 64 type: MCP
    arch: ARMv8 rev: 1 cache: L1: 512 KiB L2: 2 MiB L3: 2 MiB
  Speed (MHz): avg: 1500 min/max: 1500/2800 cores: 1: 1500 2: 1500 3: 1500
    4: 1500 bogomips: 432
  Features: Use -f option to see features
Machine:
  Type: ARM System: Pine64 RockPro64 v2.1 details: N/A
```

```
inxi -Cxx
CPU:
  Info: 6-core model: N/A variant-1: cortex-a72 variant-2: cortex-a53 bits: 64
    type: MCP arch: ARMv8 rev: 4 cache: L1: 416 KiB L2: 1.5 MiB
  Speed (MHz): avg: 1200 min/max: 408/1416:1800 cores: 1: 1200 2: 1200
    3: 1200 4: 1200 5: 1200 6: 1200 bogomips: N/A
  Features: Use -f option to see features
Machine:
  Type: ARM System: Pine64 RockPro64 v2.1 details: N/A
```

```
 inxi -Cxx
CPU:
  Info: 6-core model: N/A variant-1: cortex-a53 variant-2: cortex-a73 bits: 64
    type: MCP arch: ARMv8 rev: 4
  Speed (MHz): avg: 1000 min/max: 1000/1800:2208 cores: 1: 1000 2: 1000
    3: 1000 4: 1000 5: 1000 6: 1000 bogomips: N/A
  Features: Use -f option to see features
Machine:
  Type: ARM System: Khadas VIM3 details: N/A
```

We might want to collect `inxi -Fxxx` outputs on several ARM devices we have access to and also decide if we want to create a separate Grafana dashboard to display the collected data to avoid conflicts with the Manjaro x86_64 distribution.

**Comment by philmmanjaro** [comment]
I silent the output for now and Frede is working on aarch64 support for `get_cpuinfo`: https://github.com/manjaro/mdd/pull/2

**Comment by Dark-Sky** [comment]
rockpro64 headless via ssh so not much graphic info:

```[ray@rock64 ~]$ mdd
Welcome to MDD - The Manjaro Data Donor
Preparing data submission...
grep: /var/log/pacman.log: binary file matches

------------------------------------------
        Sending the following data
------------------------------------------
{
    "meta": {
        "version": 1,
        "timestamp": "2024-11-03T14:20:06.345160+00:00",
        "device_id": "1b44c177-f040-5bdb-a9ec-ef6a1e00e7f8",
        "distro_id": "manjaro-arm",
        "release": "24.04",
        "inxi": true
    },
    "system": {
        "kernel": "6.11.4-1-MANJARO-ARM",
        "form_factor": "single board computer",
        "install_date": "2020-04-02T15:26:13+00:00",
        "product_name": "Pine64 RockPro64 v2.1",
        "product_family": "rockpro64 rockchip",
        "sys_vendor": "pine64",
        "board_name": "rockpro64-v2.1 pine64"
    },
    "boot": {
        "uefi": false,
        "uptime_seconds": 100537
    },
    "cpu": {
        "arch": "aarch64",
        "model": "unknown",
        "cores": 6,
        "threads": 6
    },
    "memory": {
        "ram_gb": 3.771087646484375,
        "swap_gb": 5.656627655029297
    },
    "graphics": {
        "comp": "unknown",
        "dri": null,
        "gpus": [
            {
                "vendor": null,
                "model": "display-subsystem",
                "driver": "rockchip_drm"
            },
            {
                "vendor": null,
                "model": "rk3399-dw-hdmi",
                "driver": "dwhdmi_rockchip"
            },
            {
                "vendor": null,
                "model": "rk3399-mali",
                "driver": "panfrost"
            }
        ],
        "outputs": []
    },
    "audio": {
        "servers": [
            {
                "name": "PulseAudio",
                "active": true
            }
        ]
    },
    "disk": {
        "disks": [
            {
                "size_gb": 14.83740234375,
                "root": {
                    "size_gb": 14.59898328781128,
                    "fstype": "ext4",
                    "crypt": false
                },
                "home": null
            }
        ],
        "windows": false
    },
    "locale": {
        "region": "en_US.UTF-8",
        "language": "en",
        "timezone": "America/Chicago"
    },
    "package": {
        "last_update": "2022-05-30T05:54:40-05:00",
        "branch": "arm-unstable",
        "pkgs": 901,
        "foreign_pkgs": 11,
        "pkgs_update_pending": 45,
        "flatpaks": 0,
        "pacman_mirrors": {
            "total": 5,
            "ok": 5,
            "country_config": "Germany"
        }
    },
    "desktop": {
        "cli": "/bin/bash"
    }
}
------------------------------------------


```

**Comment by Dark-Sky** [comment]
vim3 headless via ssh so not much graphic info:

```
[ray@vim3 ~]$ mdd --dry-run
Welcome to MDD - The Manjaro Data Donor
Preparing data submission...

------------------------------------------
 Would send the following data (dry run)
------------------------------------------
{
    "meta": {
        "version": 1,
        "timestamp": "2024-11-03T15:03:17.482788+00:00",
        "device_id": "62782d1c-ce35-55ac-b8a9-65af227178be",
        "distro_id": "manjaro-arm",
        "release": "24.04",
        "inxi": true
    },
    "system": {
        "kernel": "6.11.4-1-MANJARO-ARM",
        "form_factor": "single board computer",
        "install_date": "2020-12-20T21:05:01+00:00",
        "product_name": "Khadas VIM3",
        "product_family": "a311d amlogic",
        "sys_vendor": "khadas",
        "board_name": "vim3 amlogic"
    },
    "boot": {
        "uefi": false,
        "uptime_seconds": 105029
    },
    "cpu": {
        "arch": "aarch64",
        "model": "unknown",
        "cores": 6,
        "threads": 6
    },
    "memory": {
        "ram_gb": 3.6921768188476562,
        "swap_gb": 5.538261413574219
    },
    "graphics": {
        "comp": "unknown",
        "dri": null,
        "gpus": [
            {
                "vendor": null,
                "model": "meson-g12a-vpu",
                "driver": "meson_drm"
            },
            {
                "vendor": null,
                "model": "meson-g12a-mali",
                "driver": "panfrost"
            },
            {
                "vendor": null,
                "model": "meson-g12a-dw-hdmi",
                "driver": "meson_dw_hdmi"
            }
        ],
        "outputs": []
    },
    "audio": {
        "servers": [
            {
                "name": "PulseAudio",
                "active": true
            },
            {
                "name": "PipeWire",
                "active": false
            }
        ]
    },
    "disk": {
        "disks": [
            {
                "size_gb": 29.12109375,
                "root": {
                    "size_gb": 28.90625,
                    "fstype": "ext4",
                    "crypt": false
                },
                "home": null
            }
        ],
        "windows": false
    },
    "locale": {
        "region": "en_US.UTF-8",
        "language": "en",
        "timezone": "America/Chicago"
    },
    "package": {
        "last_update": "2024-11-02T05:29:44-05:00",
        "branch": "arm-unstable",
        "pkgs": 1209,
        "foreign_pkgs": 24,
        "pkgs_update_pending": 7,
        "flatpaks": 0,
        "pacman_mirrors": {
            "total": 7,
            "ok": 7,
            "country_config": "Germany"
        }
    },
    "desktop": {
        "cli": "/bin/bash"
    }
}
------------------------------------------

```

**Comment by romangg** [comment]
The original issue with CPU has been fixed by #2
